Realm/Realm/Handles/AppHandle.cs

```diff
@@ -343,6 +349,23 @@ public Uri GetBaseUri()
             return new Uri(uriString);
         }
 
+        public async Task UpdateBaseUriAsync(Uri? newUri)
+        {
+            var tcs = new TaskCompletionSource();
+            var tcsHandle = GCHandle.Alloc(tcs);
+            try
+            {
+                var url = newUri?.ToString().TrimEnd('/') ?? string.Empty;
```

Comment:
What is the effect of passing a nullable Uri? Having it set to the default value?